drivers/stepper/stepper_context.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020 Teslabs Engineering S.L.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_STEPPER_STEPPER_CONTEXT_H_
+#define ZEPHYR_DRIVERS_STEPPER_STEPPER_CONTEXT_H_
+
+#include <kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct stepper_context {
+	struct k_sem lock;
+	struct k_sem sync;
+	int status;
+
+#ifdef CONFIG_STEPPER_ASYNC
+	struct k_poll_signal *signal;
```

Comment:
Can we eliminate the bool and use a non-null `signal` as a sign that the call is asynchronous?